app/controllers/ConfirmTransferController.scala

```diff
@@ -49,19 +50,33 @@ class ConfirmTransferController @Inject() (
       implicit request: Request[AnyContent]
   ): Future[Result] = {
     consignmentStatusService.getConsignmentStatuses(consignmentId, request.token.bearerAccessToken).flatMap { consignmentStatuses =>
+      val consignmentStatusValues: Map[Statuses.StatusType, Option[String]] =
+        consignmentStatusService.getStatusValues(consignmentStatuses, SeriesType, TransferAgreementType, UploadType, ClientChecksType)
       val exportTransferStatus = consignmentStatusService.getStatusValues(consignmentStatuses, ExportType).values.headOption.flatten
-      exportTransferStatus match {
-        case Some(InProgressValue.value) | Some(CompletedValue.value) | Some(FailedValue.value) =>
-          consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken).map { consignmentRef =>
-            Ok(views.html.transferAlreadyCompleted(consignmentId, consignmentRef, request.token.name)).uncache()
-          }
-        case None =>
-          getConsignmentSummary(request, consignmentId).map { consignmentSummary =>
-            httpStatus(views.html.standard.confirmTransfer(consignmentId, consignmentSummary, finalTransferForm, request.token.name)).uncache()
+      val incompleteStatuses = consignmentStatusValues.filter(s => !s._2.contains(CompletedValue.value) || s._2.contains(None))
```

Comment:
I reckon the second half of this disjunction is redundant, so we can just have
```
consignmentStatusValues.filter(s => !s._2.contains(CompletedValue.value))
```
`s._2` here is an `Option[String]`, and `contains` will evaluate as true if `s._2` equals `Some` of the argument you have given to `contains` and will evaluate as false if `s._2` ia `None`. It will never contain `None` as we're not dealing with an an `Option[Option[T]`.

I also have a slight personal preference for using `case` syntax on map tuples for readability so you can name things instead of using `._*`, so here something like 
```
      val incompleteStatuses = consignmentStatusValues.filter { case (_, statusValue) => statusValue.contains(CompletedValue.value) }
```
But your call!